plugin.video.retrospect/resources/libs/addonsettings.py

```diff
@@ -0,0 +1,1430 @@
+                 reason.
+
+        """
+
+        # This list is taken from the settings_templates.xml: geo_region
+        # 30074    |30306|30309|30308|30307|30303|30304|30301|30305|30302|30310
+        # Disabled |be   |de   |ee   |en-gb|lt   |lv   |nl   |no   |se   |dk
+        values = [None, "be", "de", "ee", "en-gb", "lt", "lv", "nl", "no", "se", "dk"]
+        value_index = AddonSettings.store(KODI).get_integer_setting("geo_region", default=0)
+        current_geograffical_region = values[value_index]
+
+        if value_only:
+            return current_geograffical_region
+
+        # if no geo region is selected, always show everything.
+        if current_geograffical_region is None:
+            return False
+
+        # only hide if the regions don't match
+        return not current_geograffical_region == channel_region
+    #endregion
+
+    #region Language caching
+    @staticmethod
+    def set_language():
+        """ Sets the language of the current Plugin run. The value is taken from the Kodi API """
+
+        language = xbmc.getLanguage()
+        if AddonSettings.__language_current != language:
+            AddonSettings.__language_strings = {}
+            Logger.info("Setting language from %s to %s", AddonSettings.__language_current, language)
+            AddonSettings.__language_current = language
+
+        return
+
+    @staticmethod
+    def get_localized_string(string_id):
+        """ returns a localized string for this id
+
+        Arguments:
+        :param int string_id: The ID for the string
+
+        :return: the localized string for this ID.
+        :rtype: str
+
+        """
+
+        translation = AddonSettings.__language_strings.get(string_id, None)
+        if translation is None:
+            translation = AddonSettings.store(KODI).get_localized_string(string_id)
+            AddonSettings.__language_strings[string_id] = translation
+
+        return translation
+    #endregion
+
+    @staticmethod
+    def send_usage_statistics():
+        """ returns true if the user allows usage statistics sending
+
+        :return: Indication if sending GA statistics is allowed.
+        :rtype: bool
+
+        """
+
+        return AddonSettings.store(KODI).get_boolean_setting("send_statistics", default=True)
+
+    @staticmethod
+    def get_current_addon_xml_md5():
+        """ Retrieves the current addons.xml.md5 content that was cached in the settings.
+
+        :return: the curreent addons.xml.md5 content
+        :rtype: str
+
+        """
+
+        return AddonSettings.store(LOCAL).get_setting(AddonSettings.__MD5_HASH_VALUE)
+
+    @staticmethod
+    def update_current_addon_xml_md5(hash_value):
+        """ Update the local cache with a new addons.xml.md5 hash value
+
+        :param str hash_value: The MD5 Hash value
+
+        """
+
+        AddonSettings.store(LOCAL).set_setting(AddonSettings.__MD5_HASH_VALUE, hash_value)
+
+    @staticmethod
+    def get_client_id():
```

Comment:
what is this used for?

Reply:
The ClientId is used as a Salt in the encryption.